sdk/identity/azure-identity/azure/identity/_authn_client.py

```diff
@@ -213,30 +214,38 @@ def request_token(
         return token
 
     def obtain_token_by_refresh_token(self, scopes, username):
-        # type: (Iterable[str], str) -> Optional[AccessToken]
+        # type: (Iterable[str], Optional[str]) -> Optional[AccessToken]
         """Acquire an access token using a cached refresh token. Returns ``None`` when that fails, or the cache has no
         refresh token. This is only used by SharedTokenCacheCredential and isn't robust enough for anything else."""
 
-        # find account matching username
-        accounts = self._cache.find(TokenCache.CredentialType.ACCOUNT, query={"username": username})
-        for account in accounts:
-            # try each refresh token that might work, return the first access token acquired
-            for token in self.get_refresh_tokens(scopes, account):
-                # currently we only support login.microsoftonline.com, which has an alias login.windows.net
-                # TODO: this must change to support sovereign clouds
-                environment = account.get("environment")
-                if not environment or (environment not in self._auth_url and environment != "login.windows.net"):
-                    continue
-
-                request = self.get_refresh_token_grant_request(token, scopes)
-                request_time = int(time.time())
-                response = self._pipeline.run(request, stream=False)
-                try:
-                    return self._deserialize_and_cache_token(
-                        response=response, scopes=scopes, request_time=request_time
-                    )
-                except ClientAuthenticationError:
-                    continue
+        #if an username is provided, restrect our search to accounts that have that username
+        query = {} if username is None else {"username": username}
+        accounts = self._cache.find(TokenCache.CredentialType.ACCOUNT, query=query)
+
+        # Only continue if we found exactly one account. We could find zero if the cache was empty (or there was
+        # no account with the specified username in the cache) and we could find more than one in the case where
+        # username was None and the cache contained multiple identities.
+        if len(accounts) != 1:
+            return None
```

Comment:
I wonder if it wouldn't be helpful to log an info level message here - at least if we bail due to the fact that there are multiple accounts and the parameter wasn't provided...

Reply:
Agreed!  Super newbie question here - what is the correct way to get an instance of a `logger` to use?  Just a module level `_LOGGING = logging.getLogger(__name__)`?  I'm not really sure what the overall logging story is.

Reply:
Per our in-person conversation - I simply added a TODO for now.